lib/Backend/Lower.cpp

```diff
@@ -20036,6 +20029,254 @@ Lowerer::GenerateCheckForCallFlagNew(IR::Instr* instrInsert)
+    IR::LabelInstr *labelProxyLoop = IR::LabelInstr::New(Js::OpCode::Label, func, false);
+    IR::LabelInstr *labelNotProxy = IR::LabelInstr::New(Js::OpCode::Label, func, false);
+
+    IR::RegOpnd *indir0RegOpnd = IR::RegOpnd::New(TyMachPtr, func);
+    IR::RegOpnd *indir1RegOpnd = IR::RegOpnd::New(TyUint32, func);
+
+    instrInsert->InsertBefore(labelProxyLoop);
+    labelProxyLoop->m_isLoopTop = true;
+
+    Loop *loop = JitAnew(func->m_alloc, Loop, func->m_alloc, this->m_func);
+    labelProxyLoop->SetLoop(loop);
+    loop->SetLoopTopInstr(labelProxyLoop);
+    loop->regAlloc.liveOnBackEdgeSyms = JitAnew(func->m_alloc, BVSparse<JitArenaAllocator>, func->m_alloc);
+    loop->regAlloc.liveOnBackEdgeSyms->Set(instanceRegOpnd->m_sym->m_id);
+
+    IR::IndirOpnd *indirOpnd = IR::IndirOpnd::New(instanceRegOpnd, Js::RecyclableObject::GetTypeOffset(), TyMachPtr, func);
+    LowererMD::CreateAssign(indir0RegOpnd, indirOpnd, instrInsert);
+
+    indirOpnd = IR::IndirOpnd::New(indir0RegOpnd, Js::Type::GetOffsetOfTypeId(), TyUint32, func);
+    LowererMD::CreateAssign(indir1RegOpnd, indirOpnd, instrInsert);
+
+    InsertCompareBranch(indir1RegOpnd, IR::IntConstOpnd::New(Js::TypeIds_Proxy, TyUint32, func, true), Js::OpCode::BrNeq_A, labelNotProxy, instrInsert);
+
+    indirOpnd = IR::IndirOpnd::New(instanceRegOpnd, Js::JavascriptProxy::GetOffsetOfTarget(), TyMachPtr, func);
+    LowererMD::CreateAssign(instanceRegOpnd, indirOpnd, instrInsert);
+
+    InsertBranch(Js::OpCode::Br, labelProxyLoop, instrInsert);
+
+    instrInsert->InsertBefore(labelNotProxy);
+
+    InsertCompareBranch(indir1RegOpnd, IR::IntConstOpnd::New(Js::TypeIds_Function, TyUint32, func, true), Js::OpCode::BrNeq_A, labelReturnFalse, instrInsert);
+
+    indirOpnd = IR::IndirOpnd::New(instanceRegOpnd, Js::JavascriptFunction::GetOffsetOfFunctionInfo(), TyMachPtr, func);
+    LowererMD::CreateAssign(indir0RegOpnd, indirOpnd, instrInsert);
+
+    indirOpnd = IR::IndirOpnd::New(indir0RegOpnd, Js::FunctionInfo::GetAttributesOffset(), TyUint32, func);
+    LowererMD::CreateAssign(indir1RegOpnd, indirOpnd, instrInsert);
+
+    InsertTestBranch(indir1RegOpnd, IR::IntConstOpnd::New(Js::FunctionInfo::Attributes::ErrorOnNew, TyUint32, func, true), Js::OpCode::BrNeq_A, labelReturnFalse, instrInsert);
+
+    InsertBranch(Js::OpCode::Br, labelReturnTrue, instrInsert);
+}
+
+void
+Lowerer::GenerateRecyclableObjectGetPrototypeNullptrGoto(IR::Instr *instrInsert, IR::RegOpnd *instanceRegOpnd, IR::LabelInstr *labelReturnNullptr)
+{
+    //  MOV instance, instance->type
+    //  MOV flags, instance->flags
+    //  TEST flags, TypeFlagMask_HasSpecialPrototype
+    //  JNE $ReturnNullptr        // external, bypassing nullptr check
+    //  MOV instance, instance->prototype
+
+    Func *func = instrInsert->m_func;
+
+    IR::RegOpnd *flagsRegOpnd = IR::RegOpnd::New(TyUint32, func);
+
+    IR::IndirOpnd *indirOpnd = IR::IndirOpnd::New(instanceRegOpnd, Js::RecyclableObject::GetTypeOffset(), TyMachPtr, func);
```

Comment:
could you use RecyclableObject::GetOffsetOfType and cleanup other uses of GetTypeOffset? There's no point having duplicate methods with different names